presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -0,0 +1,181 @@
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.StandardErrorCode.ALREADY_EXISTS;
+import static com.facebook.presto.spi.session.PropertyMetadata.stringSessionProperty;
+import static com.facebook.presto.testing.TestingSession.createBogusTestingCatalog;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestCreateTableTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private CatalogManager catalogManager;
+    private TypeManager typeManager;
+    private TransactionManager transactionManager;
+    private TablePropertyManager tablePropertyManager;
+    private Catalog testCatalog;
+    private Session testSession;
+    private MockMetadata metadata;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        catalogManager = new CatalogManager();
+        typeManager = new TypeRegistry();
+        transactionManager = createTestTransactionManager(catalogManager);
+        tablePropertyManager = new TablePropertyManager();
+        testCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        catalogManager.registerCatalog(testCatalog);
+        tablePropertyManager.addProperties(testCatalog.getConnectorId(),
+                ImmutableList.of(stringSessionProperty("baz", "test property", null, false)));
+        testSession = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(TINY_SCHEMA_NAME)
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+        metadata = new MockMetadata(typeManager,
+                tablePropertyManager,
+                testCatalog.getConnectorId());
+    }
+
+    @Test
+    public void testCreateTableNotExistsTrue()
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                true,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+        }
+        catch (RuntimeException e) {
```

Comment:
Remove the catch block and let the exception propagate normally (which fails the test)